src/Plugin/Field/FieldFormatter/VideoFormatter.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Drupal\itk_video\Plugin\Field\FieldFormatter;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\itk_video\Plugin\Field\FieldType\Video;
+use Drupal\link\Plugin\Field\FieldFormatter\LinkFormatter;
+use Drupal\itk_video\SupportedVideoProviders;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'itk_video_formatter' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "itk_video_formatter",
+ *   module = "itk_video",
+ *   label = @Translation("Video formatter"),
+ *   field_types = {
+ *     "itk_video_field"
+ *   }
+ * )
+ */
+final class VideoFormatter extends LinkFormatter {
+
+  public function __construct(
+    $plugin_id,
+    $plugin_definition,
+    FieldDefinitionInterface $field_definition,
+    array $settings,
+    $label,
+    $view_mode,
+    array $third_party_settings,
+    protected $urlResolver,
+    protected $httpClient,
+    // @phpstan-ignore property.phpDocType
+    protected $pathValidator,
+    protected ConfigFactoryInterface $configFactory,
+  ) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $label, $view_mode, $third_party_settings, $pathValidator);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+  ): VideoFormatter {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['label'],
+      $configuration['view_mode'],
+      $configuration['third_party_settings'],
+      $container->get('media.oembed.url_resolver'),
+      $container->get('http_client'),
+      $container->get('path.validator'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode): array {
+    $elements['#attached']['library'][] = 'itk_video/video';
+
+    foreach ($items as $delta => $item) {
+
+      $this->createVideo($item);
+
+      if (!empty($item->getUrl()->toString())) {
+        $markup =
+          '<div class="itk-video itk-video-responsive">' . $this->createVideo($item) . '</div>';
+
+        $elements[$delta] = [
+          '#type' => 'inline_template',
+          '#template' => $markup,
+        ];
```

Comment:
Are you using `inline_template` to prevent Drupal from escaping the HTML returned from `createVideo`? It may be easier to read and understand

```suggestion
        $elements[$delta] = [
          '#prefix' => '<div class="itk-video itk-video-responsive">',
          '#markup' => $this->createVideo($item),
          '#suffix' => '</div>',
        ];
```

but I'm not sure if Drupal escapes HTML in `#markup`.

Reply:
Yeah tested it and #markup filters out the iframe tag. I changed it to inline_template while maintaining the prefix/suffix part